etstool.py

```diff
@@ -0,0 +1,334 @@
+@click.option(
+    '--environment', default=None, help='Override the default environment name')
+def test(runtime, toolkit, environment):
+    """ Run the test suite in a given environment with the specified toolkit.
+
+    """
+    parameters = get_parameters(runtime, toolkit, environment)
+    environ = environment_vars.get(toolkit, {}).copy()
+    environ['PYTHONUNBUFFERED'] = "1"
+    commands = [
+        "edm run -e {environment} -- coverage run -p -m nose.core -v traits_enaml --nologcapture"]
+
+    # We run in a tempdir to avoid accidentally picking up wrong traitsui
+    # code from a local dir.  We need to ensure a good .coveragerc is in
+    # that directory, plus coverage has a bug that means a non-local coverage
+    # file doesn't get populated correctly.
+    click.echo("Running tests in '{environment}'".format(**parameters))
+    with do_in_tempdir(files=['.coveragerc'], capture_files=['./.coverage*']):
+        os.environ.update(environ)
+        execute(commands, parameters)
+    execute(["edm run -e {environment} -- coverage report"], parameters)
+    click.echo('Done test')
+
+
+@cli.command()
+@click.option('--runtime', default='2.7', help='The python runtime version')
+@click.option(
+    '--toolkit', type=click.Choice(['pyside', 'pyqt4']),
+    default='pyside', help='The gui toolkit to use')
+@click.option(
+    '--environment', default=None, help='Override the default environment name')
+def cleanup(runtime, toolkit, environment):
+    """ Remove a development environment.
+
+    """
+    parameters = get_parameters(runtime, toolkit, environment)
+    commands = [
+        "edm run -e {environment} -- python setup.py clean",
+        "edm environments remove {environment} --purge -y"]
+    click.echo("Cleaning up environment '{environment}'".format(**parameters))
+    execute(commands, parameters)
+    click.echo('Done cleanup')
+
+
+@cli.command()
+@click.option('--runtime', default='2.7', help='The python runtime version')
+@click.option(
+    '--toolkit', type=click.Choice(['pyside', 'pyqt4']),
+    default='pyside', help='The gui toolkit to use')
+@click.option(
+    '--environment', default=None, help='Override the default environment name')
+def test_clean(runtime, toolkit):
+    """ Run tests in a clean environment, cleaning up afterwards
+
+    """
+    args = ['--toolkit={}'.format(toolkit), '--runtime={}'.format(runtime)]
+    try:
+        install(args=args, standalone_mode=False)
+        test(args=args, standalone_mode=False)
+    finally:
+        cleanup(args=args, standalone_mode=False)
+
+
+@cli.command()
+@click.option('--runtime', default='2.7', help='The python runtime version')
+@click.option(
+    '--toolkit', type=click.Choice(['pyside', 'pyqt4']),
+    default='pyside', help='The gui toolkit to use')
+@click.option(
+    '--environment', default=None, help='Override the default environment name')
+def update(runtime, toolkit, environment):
+    """ Update/Reinstall package into environment.
```

Comment:
Is the main argument for this command that you can install the package into multiple environments simultaneously? I can imagine `setup.py develop` being a nightmare when straddling Python 2 & 3.

Reply:
Yes it is a convinience function that I found useful while working on libraries that need to support multiple combinations of python versions and toolkits